mobile/library/objective-c/EnvoyConfiguration.mm

```diff
@@ -255,13 +255,21 @@ - (instancetype)initWithAdminInterfaceEnabled:(BOOL)adminInterfaceEnabled
     }
     builder.addStatsSinks(std::move(sinks));
   }
-  builder.setNodeLocality([self.nodeRegion toCXXString], [self.nodeZone toCXXString],
-                          [self.nodeSubZone toCXXString]);
-  builder.setNodeId([self.nodeId toCXXString]);
-  builder.addRtdsLayer([self.rtdsLayerName toCXXString], self.rtdsTimeoutSeconds);
-  builder.setAggregatedDiscoveryService(
-      [self.adsAddress toCXXString], self.adsPort, [self.adsJwtToken toCXXString],
-      self.adsJwtTokenLifetimeSeconds, [self.adsSslRootCerts toCXXString]);
+  if (self.nodeRegion != nil) {
+    builder.setNodeLocality([self.nodeRegion toCXXString], [self.nodeZone toCXXString],
+                            [self.nodeSubZone toCXXString]);
+  }
+  if (self.nodeId != nil) {
+    builder.setNodeId([self.nodeId toCXXString]);
+  }
+  if (self.rtdsLayerName != nil) {
```

Comment:
a bit off that ordering in obj-c files is different than in JNI files but I guess it is what it is. Given that we have 2 places that need to implement it there is no way to ensure that they follow the same ordering.

Reply:
Yeah that shouldn't matter since they're still built at the same place.